nifi-nar-bundles/nifi-soap-bundle/nifi-soap-processors/src/main/java/org/apache/nifi/processors/soap/GetSOAP.java

```diff
@@ -0,0 +1,508 @@
+import org.apache.nifi.components.ValidationResult;
+import org.apache.nifi.components.Validator;
+import org.apache.nifi.flowfile.FlowFile;
+import org.apache.nifi.flowfile.attributes.CoreAttributes;
+import org.apache.nifi.logging.ComponentLog;
+import org.apache.nifi.processor.AbstractProcessor;
+import org.apache.nifi.processor.ProcessContext;
+import org.apache.nifi.processor.ProcessSession;
+import org.apache.nifi.processor.ProcessorInitializationContext;
+import org.apache.nifi.processor.Relationship;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.processor.util.StandardValidators;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+@SupportsBatching
+@InputRequirement(InputRequirement.Requirement.INPUT_ALLOWED)
+@Tags({"SOAP", "Get", "Ingest", "Ingress"})
+@CapabilityDescription(
+        "Execute provided request against the SOAP endpoint. The result will be left in it's orginal form and all " +
+        "headers that are returned will be written as attributes to the resulting flow file.")
+@WritesAttributes({@WritesAttribute(attribute = "getsoap.status.code", description = "The status code that is returned"),
+        @WritesAttribute(attribute = "getsoap.status.message", description = "The status message that is returned"),
+        @WritesAttribute(attribute = "getsoap.request.wsdl_url", description = "The request WSDL URL"),
+        @WritesAttribute(attribute = "getsoap.request.method", description = "The request method"),
+        @WritesAttribute(attribute = "mime.type", description = "Sets mime type to text/xml")})
+@DynamicProperty(name = "The name of a input parameter the needs to be passed to the SOAP method being invoked.",
+        value = "The value for this parameter '=' and ',' are not considered valid values and must be escaped . " +
+                "Note, if the value of parameter needs to be an array the format should be key1=value1,key2=value2.  ",
+        description = "The name provided will be the name sent in the SOAP method, therefore please make sure " +
+                      "it matches the wsdl documentation for the SOAP service being called. In the case of arrays " +
+                      "the name will be the name of the array and the key's specified in the value will be the element " +
+                      "names passed.")
+
+public class GetSOAP extends AbstractProcessor {
+
+    protected static final PropertyDescriptor ENDPOINT_URL = new PropertyDescriptor
+            .Builder()
+            .name("endpoint-url")
+            .displayName("Endpoint URL")
+            .description("The endpoint url that hosts the web service(s) that should be called.")
+            .required(true)
+            .expressionLanguageSupported(false)
```

Comment:
I think expression language should be supported on almost all the properties: with the variable registry, it can be used to externalize values that would be different between environments, also, since this processor can accept incoming flow files, this could be read from the attributes of the flow files.